cabal-install/Distribution/Client/IndexUtils.hs

```diff
@@ -179,6 +180,9 @@ readRepoIndex verbosity repo mode =
           RepoRemote{..} -> warn verbosity $
                "The package list for '" ++ remoteRepoName repoRemote
             ++ "' does not exist. Run 'cabal update' to download it."
+          RepoSecure{..} -> warn verbosity $
```

Comment:
I had in fact originally done that but when I tried hard to show that the new secure stuff didn't affect any existing code paths I changed it back to this :)